File: ViewletCreatorIOS/ViewletCreator/Classes/utility/ViewletDimensionLookup.swift
//
//  ViewletDimensionLookup.swift
//  Viewlet creator Pod
//
//  Library utility: protocol for dimension lookup
//  Integrates with ViewletConvUtil to look up custom dimension sizes or coordinates
//

import UIKit

public protocol ViewletDimensionLookup {
    
    func getDimension(refId: String) -> CGFloat?
    
}

File: ViewletCreatorAndroid/ViewletCreatorLib/src/main/java/com/crescentflare/viewletcreator/utility/ViewletDimensionLookup.java
package com.crescentflare.viewletcreator.utility;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

/**
 * Viewlet creator utility: interface for dimension lookup
 * Integrates with ViewletMapUtil to look up custom dimension references
 */
@SuppressWarnings({"unused", "WeakerAccess"})
public interface ViewletDimensionLookup
{
    @Nullable
    Integer getDimension(@NotNull String refId);
}
